src/overlays/actors/ovl_En_Guard_Nuts/z_en_guard_nuts.c

```diff
@@ -28,42 +36,338 @@ const ActorInit En_Guard_Nuts_InitVars = {
+                func_801477B4(globalCtx);
+                this->unk21C = 4;
+                this->actionFunc = EnGuardNuts_Unburrow;
+            } else {
+                func_801477B4(globalCtx);
+                EnGuardNuts_SetupWait(this);
+            }
+        }
+    } else if ((Message_GetState(&globalCtx->msgCtx) >= 3) && (D_80ABBE20 == 0)) {
+        if ((this->textId == 0) || (this->textId == 3) || (this->textId >= 7)) {
+            if (this->timer == 0) {
+                this->timer = 2;
+                this->unk23A.y ^= 0x2000;
+            }
+        } else if (this->timer == 0) {
+            this->timer = 2;
+            this->unk23A.x ^= 0x2000;
+        }
+    }
+}
+
+void EnGuardNuts_Burrow(EnGuardNuts* this, GlobalContext* globalCtx) {
+    Vec3f digPos;
+
+    EnGuardNuts_ChangeAnim(this, DIG_ANIM);
+    Math_Vec3f_Copy(&digPos, &this->actor.world.pos);
+    digPos.y = this->actor.floorHeight;
+    EffectSsHahen_SpawnBurst(globalCtx, &digPos, 4.0f, 0, 10, 3, 15, -1, 10, NULL);
+    this->unk23A.y = 0;
+    this->actor.flags |= ACTOR_FLAG_8000000;
+    this->unk23A.x = this->unk23A.y;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_NUTS_DOWN);
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_NUTS_UP);
+    this->actionFunc = EnGuardNuts_SetupUnburrow;
+}
+
+void EnGuardNuts_SetupUnburrow(EnGuardNuts* this, GlobalContext* globalCtx) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (this->animFrameCount <= curFrame) {
+        this->unk21C = 3;
+        this->actionFunc = EnGuardNuts_Unburrow;
+        this->actor.world.rot.y = this->actor.home.rot.y;
+        this->actor.shape.rot.y = this->actor.home.rot.y;
+    }
+}
+
+void EnGuardNuts_Unburrow(EnGuardNuts* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+    Vec3f digPos;
+
+    // If you have returned Deku Princess, guards will not unburrow
+    if (!(gSaveContext.weekEventReg[23] & 0x20)) {
+        yawDiff = ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.home.rot.y));
+        if ((yawDiff < 0x4000) && ((D_80ABBE20 == 0) || (this->actor.xzDistToPlayer > 150.0f))) {
+            Math_Vec3f_Copy(&digPos, &this->actor.world.pos);
+            digPos.y = this->actor.floorHeight;
+            EffectSsHahen_SpawnBurst(globalCtx, &digPos, 4.0f, 0, 10, 3, 15, -1, 10, NULL);
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_NUTS_UP);
+            D_80ABBE20 = 0;
+            if (this->textId == 9) {
+                this->hasCompletedConversation = true;
+            }
+            this->actor.flags &= ~ACTOR_FLAG_8000000;
+            EnGuardNuts_SetupWait(this);
+        }
+    }
+}
 
-extern ColliderCylinderInit D_80ABBDF0;
+void EnGuardNuts_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuardNuts* this = THIS;
+    s32 pad;
 
-extern UNK_TYPE D_06002700;
-extern UNK_TYPE D_06002A08;
+    if (this->blinkTimer == 0) {
+        this->eyeState++;
+        if (this->eyeState >= 3) {
```

Comment:
You don't have to do this, but these two back to back can often be combined into
`if (++this->eyeState >= 3) { `

Just thought I'd point that out